### PR TITLE
ci: use uv, require recent nox

### PR DESCRIPTION
**Pull request by henryiii**
See how this fares.

### CODE REVIEW

**Comment by henryiii** [comment]
Checking a few job times: 3.8 on macOS went from 6 minutes to 2 minutes. Check included noxfile went from 7 minutes to 3 minutes. 3.12 on Windows went from 9 minutes to 5 minutes. 3.8 on Windows went from 8 minutes to 6 minutes.

File: docs/pages/tutorials/dev-environment.md [hunk 1]
```diff
@@ -87,6 +87,16 @@ leave the environment (or just close your shell).
 > If you like a different shell, like fish, there are several `activate`
 > scripts; the default one expects a bash-like shell.
 
+{: .note-title }
+
+> Fast alternative
+>
+> You can also consider the [uv][] package, which is much, much faster version
+> of pip and venv implemented in Rust. Just put `uv` in front of the commands
+> you'd normally use; as long as use use venvs, it should be nearly the same.
```
Comment (henryiii) [inline docs/pages/tutorials/dev-environment.md:96]:
```suggestion
> you'd normally use; as long as you use venvs, it should be nearly the same.
```

File: docs/pages/guides/gha_basic.md [hunk 2]
```diff
@@ -276,6 +285,8 @@ branches that you can use.
   Publish Python packages to PyPI. Supports trusted publisher deployment.
 - [re-actors/alls-green](https://github.com/re-actors/alls-green): Tooling to
   check to see if all jobs passed (supports allowed failures, too).
+- [sigstore/gh-action-sigstore-python](https://github.com/sigstore/gh-action-sigstore-python):
```
Comment (webknjaz) [inline docs/pages/guides/gha_basic.md:288]: @henryiii FYI the sigstore action does not have a `release/v1` branch. This is the pattern that I tend to follow, but that action isn't mine and William does not use it.

Comment (henryiii) [reply docs/pages/guides/gha_basic.md:288]: Oops, I missed the heading. I can move it up. Thanks!

**Pull request merged**